vortex-duckdb/src/convert/array/mod.rs

```diff
@@ -57,36 +57,54 @@ pub fn to_duckdb(
     chunk: &mut dyn WritableVector,
     cache: &mut ConversionCache,
 ) -> VortexResult<()> {
+    if try_to_duckdb(array, chunk, cache)?.is_some() {
+        return Ok(());
+    };
+    let canonical_array = array.to_canonical()?.into_array();
+    if try_to_duckdb(&canonical_array, chunk, cache)?.is_some() {
+        return Ok(());
+    };
+    to_arrow_preferred(&canonical_array)?.to_duckdb(chunk, cache)
+}
+
+fn try_to_duckdb(
+    array: &ArrayRef,
+    chunk: &mut dyn WritableVector,
+    cache: &mut ConversionCache,
+) -> VortexResult<Option<()>> {
```

Comment:
Why not map everything to `VortexResult<()>` without the inner `Option`?

Reply:
That is lossy

Reply:
I want to know if there was a to_duck fn to run or not, that is what the option is